### PR TITLE
doctl: init at 1.3.1

### PR DESCRIPTION
**Pull request by langston-barrett**
###### Motivation for this change

[DigitalOcean](https://www.digitalocean.com/) is one of the biggest cloud hosting services around, coming in right behind Amazon Web Services and Google Compute Engine. This is the CLI wrapper for their API. 
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/doctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  version = "1.3.1";
+  rev = "a57555c195d06bc7aa5037af77fde0665ad1231f";
+  goPackagePath = "github.com/digitalocean/doctl";
+
+  src = fetchFromGitHub {
+    owner = "digitalocean";
+    repo = "doctl";
+    rev = "${rev}";
+    sha256 = "03z652fw0a628gv666w8vpi05a4sdilvs1j5scjhcbi82zsbkvma";
+  };
```
Comment (kamilchm) [inline pkgs/development/tools/doctl/default.nix]: meta?

Comment (siddharthist) [inline pkgs/development/tools/doctl/default.nix]: What does that mean?

Comment (joachifm) [inline pkgs/development/tools/doctl/default.nix]: @siddharthist there's no license info, homepage, etc., see https://nixos.org/nixpkgs/manual/#chap-meta

Comment (siddharthist) [inline pkgs/development/tools/doctl/default.nix]: @joachifm Thanks for the info, I'll add it!

Comment (siddharthist) [inline pkgs/development/tools/doctl/default.nix]: @joachifm @kamilchm done :smile:

File: lib/maintainers.nix [hunk 2]
```diff
@@ -356,6 +356,7 @@
   sheganinans = "Aistis Raulinaitis <sheganinans@gmail.com>";
   shell = "Shell Turner <cam.turn@gmail.com>";
   shlevy = "Shea Levy <shea@shealevy.com>";
+  siddharthist = "Langston Barrett <langston.barrett@gmail.com>"
```
Comment (fpletz) [inline lib/maintainers.nix]: There's a semicolon missing at EOL.

Comment (siddharthist) [inline lib/maintainers.nix]: @fpletz Good eye, thanks! fixed now.

File: pkgs/development/tools/doctl/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = {
+    description = "A command line tool for DigitalOcean services";
+    homepage = "https://github.com/digitalocean/doctl";
+    license = stdenv.lib.licenses.apache2;
```
Comment (fpletz) [inline pkgs/development/tools/doctl/default.nix]: Sorry, another one. The Apache 2.0 license is called `asl20` in `lib/licenses.nix`. Travis fails due to this.

Comment (siddharthist) [inline pkgs/development/tools/doctl/default.nix]: @fpletz not at all, thanks for the thorough review! Sorry, I couldn't get `nix-build` to build a new package after adding the `meta` section (which is why there are obvious unbuildable errors). Do you have any suggestions for forcing a new build?

Comment (siddharthist) [inline pkgs/development/tools/doctl/default.nix]: @fpletz (also, fixed)

Comment (joachifm) [inline pkgs/development/tools/doctl/default.nix]: The `meta` attribute is not taken into account when computing the output hash, so changes to `meta` won't trigger a rebuild. You can do
`nix-instantiate --eval --strict -A foo.meta` to force evaluation of the `meta` attribute.

**Comment by fpletz** [comment]
Hmm, travis has vanished. Seems we managed to break Github somehow. :beers:

**Comment by fpletz** [comment]
Built & tested locally. Squashed your commits and pushed to master.

**Comment by siddharthist** [comment]
Thanks to all for the help & review!

**Pull request merged**